authClient/authClient.go

```diff
@@ -66,9 +67,15 @@ func (c *E3dbAuthClient) AuthenticateE3DBClient(ctx context.Context, token strin
 
 // AuthHTTPClient returns an http client that can be used for
 // making authenticated requests to an e3db endpoint using the provided context.
+// The first time AuthHTTPClient is called it use the provided context to create a new http client
+// and will reuse that context until it has expired
```

Comment:
This still reuses contexts from requests for other requests, which seems off. Should this just use a background context instead?

Reply:
Thats actually explicitly how we're using context here, the top level request sets the context and passes it down to every other function that is processing the request, allowing the request to be cancelled upstream and have any downstream work stop. Using a background context here would be some implicit magic that would break the contract expected by a user of this library(e.g. I give you context, you use context)